### ISSUE TITLE
Add useful slit/source-related keywords to combine1d headers

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5605]
_Issue [JP-1841](https://jira.stsci.edu/browse/JP-1841) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

The table extension header in combine_1d (c1d) output products currently doesn't contain any of the slit/source-related keywords that are available in the extension headers of the input x1d products. Some of these could be useful to users and hence should be copied over. Right now the step is setup to only copy meta data from the primary HDU when creating the c1d output product.

At the same time we should think about possibly changing the EXTNAME's of the table extensions in both x1d and c1d products from their current values (EXTRACT1D, COMBINE1D) to the generic value "SCI". This would greatly simplify schema construction (making it possible to reuse schemas for multiple data models) and would make the naming consistent with other science products.

**Comment by jdavies-st** [comment]
Agree that reusing the names would be great!

Also, calling the tables `data` in the datamodel would simplify finding the correct items too, instead of trying to remember what this or that datamodel called it's primary data array.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-1841?focusedCommentId=509485#comment-509485):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) In looking at this, I think I might need a bit of clarification on what exactly is requested. For each x1d header, I see these relevant keywords:

```python
SLITID  =                    0 / Slitlet ID
SOURCEID=                    1 / Source ID
SRCXPOS =    2021.425373012163 / Source position in slit (x-axis)
SRCYPOS =    7.889378008887661 / Source position in slit (y-axis)
SLIT_RA =    90.81187087093895 / Right ascension (deg) at middle of slit
SLIT_DEC=   -66.56426910346784 / Declination (deg) at middle of slit
SPORDER =                    1 / Spectral order number
DISPAXIS=                    2 / Dispersion direction
```

The spectral order is already in the c1d headers, as it's a constant value across the combined x1d input tables. For some of the other keywords, though, the values are x1d-dependent amongst the input x1ds to a single c1d output. Should I avoid providing these keywords, or is it intended that I choose the first/average/? value for these keywords?

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1841?focusedCommentId=509584#comment-509584):_

Theoretically we should only ever be combining spectra from the same target and same spectral order, so some keyword values should be the same for all input instances, but not necessarily all of them. This method is primarily used to combine spectra from modes that do not undergo resampling (and hence can't be combined in 2D space), which primarily only includes NIRCam and NIRISS WFSS. Other modes like NIRCam Time Series Grism, NIRISS SOSS, and MIRI LRS slitless, are typically handled as TSO cases, which means spectra are never combined (just concatenated into a time series).

For the WFSS modes, values like SRCXPOS/SRCYPOS could change due to the use of dithering between exposures. But because it's the same source, one would expect SLIT_RA/SLIT_DEC to remain constant (or very nearly so) across exposures. I believe SOURCEID comes from the source number in the source_cat derived from the associated direct image of the field and because there's only 1 source_cat driving the processing of all grism exposures in a given observation, the SOURCEID should also remain the same for a given source across multiple exposures. I'm not sure about SLITID. We'd have to check the code in /extract_2d/grisms.py to see where that value comes from, in order to figure out if it might change from exposure to exposure for a given source.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-1841?focusedCommentId=510430#comment-510430):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) It looks as though the slit name (which I left out of my snippet above) is set with the same value as the SOURCEID keyword. The SLITID doesn't appear to be relevant in WFSS modes, and the SRCPOS/SLITRA pairs change between x1d exposures, and so are not constant across the c1d combination. From this, I'm taking that the c1d headers should include SPORDER and SOURCEID. Do you agree that this is sufficient?

Edit: should I also be modifying the FITS extension names to SCI in this ticket, or is that a separate suggestion?

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1841?focusedCommentId=510616#comment-510616):_

Fixed by #5759
